packages/firestore/src/api/database.ts

```diff
@@ -194,32 +194,15 @@ class FirestoreSettings {
     // Nobody should set timestampsInSnapshots anymore, but the error depends on
     // whether they set it to true or false...
     if (settings.timestampsInSnapshots === true) {
-      logError(`
-  The timestampsInSnapshots setting now defaults to true and you no
-  longer need to explicitly set it. In a future release, the setting
-  will be removed entirely and so it is recommended that you remove it
-  from your firestore.settings() call now.`);
+      logError(
+        "The setting 'timestampsInSnapshots: true' is no longer required " +
+          'and should be removed.'
+      );
     } else if (settings.timestampsInSnapshots === false) {
-      logError(`
-  The timestampsInSnapshots setting will soon be removed. YOU MUST UPDATE
-  YOUR CODE.
-
-  To hide this warning, stop using the timestampsInSnapshots setting in your
-  firestore.settings({ ... }) call.
-
-  Once you remove the setting, Timestamps stored in Cloud Firestore will be
-  read back as Firebase Timestamp objects instead of as system Date objects.
-  So you will also need to update code expecting a Date to instead expect a
-  Timestamp. For example:
-
-  // Old:
-  const date = snapshot.get('created_at');
-  // New:
-  const timestamp = snapshot.get('created_at'); const date =
-  timestamp.toDate();
-
-  Please audit all existing usages of Date when you enable the new
-  behavior.`);
+      logError(
+        "Support for 'timestampsInSnapshots: false' will be removed soon. " +
+          'You must update your code to handle Timestamp objects.'
```

Comment:
What do you think of putting the helpful text about how to handle this in a file like `packages/firestore/deprecation.md` and then linked to it? Then this could be

```ts
"Support for 'timestampsInSnapshots: false' will be removed. " +
  'See https://github.com/firebase/firebase-js-sdk/packages/firestore/deprecation.md#timestamps'
```

It's almost as short but remains just as helpful.

Reply:
I expanded on the comment in our API docs and added a GIST with the comment (since we can't permalink to the API doc).

Reply:
This is probably better linked to a file actually in the repo rather than a personal gist, similar to what we did here: https://github.com/firebase/firebase-android-sdk/blob/fa9a8c78ad8cae66dec02da9edb40c5303286cc8/firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AbstractStream.java#L282

Reply:
FWIW I went back to the original error message (but left the updates to the API documentation). This error message is so old that the number of people that see this probably does not warrant any extra steps.